src/System.Net.HttpListener/tests/HttpListenerAuthenticationTest.cs

```diff
@@ -384,22 +390,21 @@ public void Realm_SetDisposed_ThrowsObjectDisposedException()
 
         public async Task<HttpResponseMessage> AuthenticationFailure(HttpClient client, HttpStatusCode errorCode)
         {
+            client.Timeout = new TimeSpan(0, 0, 1);
             Task<HttpResponseMessage> clientTask = client.GetAsync(_factory.ListeningUrl);
 
             // The server task will hang forever if it is not cancelled.
             var tokenSource = new CancellationTokenSource();
             Task<HttpListenerContext> serverTask = Task.Run(() => _listener.GetContext(), tokenSource.Token);
-
+            
             // The client task should complete first - the server should send a 401 response.
             Task resultTask = await Task.WhenAny(clientTask, serverTask);
             tokenSource.Cancel();
             if (resultTask == serverTask)
             {
-                await serverTask;
+                await clientTask;
```

Comment:
> Why did you change this?

Awaiting a task that has finished is always a bug in my mind. The code written above will trigger the compiler to generate an iterator class and function which isn't necessary if we already know the task is done. I don't see how the compiler can optimize this away given the resultTask may not be equal to serverTask.

> If the WhenAny completes due to the serverTask completing, then something went very wrong, and we want to await the serverTask to propagate any exception it may have had. 

If it's _very_ wrong we shouldn't propagate the exception. Instead we should crash immediately with Debug.Fail and get a dump to better understand the state of both the client and the server (including native code).
Otherwise we could propagate with GetAwaiter.GetResult() instead of the await and maybe have a descriptive Assert statement or comment with the text that you wrote.